src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -87,6 +89,7 @@ public static void Initialize()
             {
                 // Initialize SdkSelfDiagnosticsEventListener to create an EventListener for the OpenTelemetry SDK
                 _sdkEventListener = new(EventLevel.Warning, Logger);
+                _lazyInstrumentation = new(new LazyInstrumentationBuilders(TracerSettings));
```

Comment:
This looks strange to me. Once you build the `LazyInstrumentationBuilders` and use it in the constructor of the `LazyInstrumentationLoader`, you just ignore it. Could you instead combine all of the logic from `LazyInstrumentationBuilders` into the `LazyInstrumentationLoader` class and only create that object?

Reply:
The concrete implementations were separated and moved to `LazyInstrumentationBuilders` to support testing. Source where the modification comes in: https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/pull/903#discussion_r911803125